PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc

```diff
@@ -221,6 +237,14 @@ class GenWeightsTableProducer : public edm::global::EDProducer<edm::StreamCache<
                 if (mNamed != namedWeightIDs_.end()) wNamed[mNamed-namedWeightIDs_.begin()] = weight.wgt/w0;
             } 
 
+            int vectorSize = genProd.weights().size() > 1 ? genProd.weights().size()-2 : 0;
```

Comment:
with the present logic, it might be safer to check whether `size() == 14` instead of `size() > 1`, as we don't really know what is the interpretation of the PS weights if they're not 2+12.
that way, if there is some sample produced with different ps weights settings we'll end up not storing anything (and people trying to access it will notice it), while with the current code we'd be still storing weights but they may not be the right ones, and people could get bogus results.